### PR TITLE
[CPU] Split supported PD initialization refactoring plus tests for multiply port connections.

### PR DESCRIPTION
**Pull request by maxnick**
### Details:
Supported primitive descriptors initialization has been rewritten using TensorDescCreators to make this procedure aligned with the other layers. Also necessary tests that check the case when multiple edges connected to the same port was added.

### Tickets:
 - 49925

### CODE REVIEW

**Comment by maxnick** [comment]
@dmitry-gorokhov could you please review?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp [hunk 1]
```diff
@@ -362,9 +335,16 @@ void MKLDNNSplitNode::initOptimalPrimitiveDescriptor() {
 }
 
 void MKLDNNSplitNode::selectOptimalPrimitiveDescriptor() {
-    if (implPriorities.size() > 0 && implPriorities[0] == impl_desc_type::ref) {
-        selectPrimitiveDescriptorByIndex(0);
-        return;
+    if (!implPriorities.empty() && implPriorities[0] == impl_desc_type::ref) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:338]: Could you please add the comment why we need special case here

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp [hunk 2]
```diff
@@ -122,15 +81,42 @@ void MKLDNNSplitNode::initSupportedPrimitiveDescriptors() {
...
         dynBatchSupport = false;
     }
-    auto makePdInfo = [dynBatchSupport](TensorDescFactory getTensorDesc, const Precision& precision,  const MKLDNNDims& srcDims,
-                                        const std::vector<MKLDNNDims>& outDims, impl_desc_type type) -> PrimitiveDescInfo {
+
+    //Set plain and taiC formats
+    std::vector<TensorDescCreatorTypes> tdCreatorTypes{ TensorDescCreatorTypes::ncsp, TensorDescCreatorTypes::nspc };
+
+    //Support channel blocked format
+    if (srcDims.ndims() > channelsPos) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:89]: Does it really make sense to create blocked descs for 2D inputs?

**Comment by dmitry-gorokhov** [comment]
@maxnick  In general LGTM. Please fix remaining comments and rebase on latest master to get fresh CI.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp (hunk 1 above)
Comment (maxnick) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:339]: Done

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp (hunk 2 above)
Comment (maxnick) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp]: In an arbitrary case yes. But considering real-world applications, it seems needless. The threshold has been changed to the rank of three bearing in mind possible native support for 1d convolutions.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**